### DISCUSSION TITLE
Python Fact suggestions

### DISCUSSION THREAD

**Discussion by jb3** [python-discord/meta#93] [General]
Please put suggestions for our `.pyfact` command here!

Any fun facts about Python can be submitted.

All current facts are located in the Sir Lancebot repository [here](https://github.com/python-discord/sir-lancebot/blob/master/bot/resources/evergreen/python_facts.txt).

**Comment by UP929312** [comment]
Fact: 
The name "Python" is a tribute to the British comedy group, "Monty Python", and is often mentioned in tutorials and reference materials, such as referring to spam and eggs (from a famous Monty Python sketch) instead of the standard "foo" and "bar".

**Comment by UP929312** [comment]
Fact:
Python was first released in 1991, a few years before Java even!

**Comment by IFcoltransG** [comment]
Python was invented by Guido Van Rossum, who was affectionately known as the 'Benevolent Dictator For Life' of Python (BDFL) until he stepped down in 2018.

**Comment by MrKomodoDragon** [comment]
If you type `from __future__ import braces` in the REPL, you’ll get an interesting SyntaxError about whether it will be implemented.

**Comment by Kronifer** [comment]
Not only did Guido van Rossum name Python after Monty Python, the Python docs are riddled with Monty Python skit references.

**Comment by Brodevil** [comment]
"Python is a multi-paradigm programming language providing support for procedural, object-oriented, and even functional paradigms".
It's a bit hard to quantify all paradigms like such as Python itself is an extremely flexible language

1. Structured programming
2. Functional programming
3. Object-Oriented programming (OOPs)

**Reply by purefunctor** [reply]
I would probably phrase this as "Python is a multi-paradigm programming language providing support for procedural, object-oriented, and even functional paradigms".

It's a bit hard to quantify all paradigms like such as Python itself is an extremely flexible language

**Comment by Brodevil** [comment]
A bank python file is also having some inbuild hiden identifiers, by Default
That are :
```py
__doc__
__name__
__cached__
__sepc__
__loader__
__pakage__
__annotations__
___file___   
__buildins__
__bader__
```
These are containing some data as per their works.

**Reply by swfarnsworth** [reply]
What do you mean by "a bank python file"?

**Reply by Brodevil** [reply]
A python file which doesn't contain any data

**Reply by 15696** [reply]
I would rephrase this as:

> A blank Python file has some built-in hidden variables by default. These are:
```
__doc__
__name__
__cached__
__sepc__
__loader__
__package__
__annotations__
__file__
__buildins__
__bader__
```

I'm not too familiar with these as I don't use them often but I think there might be some spelling mistakes, like `__buildins__` which I'm pretty sure should be `__builtins__`.

**Comment by Brodevil** [comment]
Python Libraries and Packages are a set of useful modules and functions that minimize the use of code in our day-to-day life. 
There are over more than `137,000 python libraries` and `198,826 python packages ` ready to ease the developer's regular programming experience.
These libraries and packages are intended for a variety of modern-day solutions.

**Reply by 15696** [reply]
I don't think stating what a library / package is is necessary.

Maybe just something like:

> There are more than **137k libraries** and **198k packages** in Python.

**Comment by Kronifer** [comment]
There are many implementations of Python, like Jython which runs on the Java Virtual Machine, iron python on .net and pypy which uses a just in time compiler are just a few

**Comment by Kronifer** [comment]
There is an unofficial port of python for the PS2

**Reply by UP929312** [reply]
Source?

**Reply by purefunctor** [reply]
This mailing list thread, it seems https://mail.python.org/pipermail/python-dev/2001-July/016365.html

**Comment by thatbirdguythatuknownot** [comment]
There is a module named `turtledemo` that animates and creates things with the `turtle` module. An example animation is `turtledemo.sorting_animate`, which can be tested using `from turtledemo.sorting_animate import main; main()`.

**Comment by thatbirdguythatuknownot** [comment]
When you do `hash(float('inf'))` or `hash(math.inf)`, you'll get the first few digits of pi but as an int: `314159`.

**Comment by greyblue9** [comment]
Did you know the GOAT of python, Ned, commonly hangs around and help users at python discord!!

**Reply by nedbat** [reply]
> Did you know the GOAT of python, Ned, commonly hangs around and help users at python discord!!

I don't think we need to add this as a fact :)

**Reply by greyblue9** [reply]
> > Did you know the GOAT of python, Ned, commonly hangs around and help users at python discord!!
> 
> I don't think we need to add this as a fact :)

Removed from PR :D

**Comment by greyblue9** [comment]
Apparently, pythons are known to have eaten goats.
